src/vs/editor/contrib/hover/browser/modesContentHover.ts

```diff
@@ -245,24 +242,7 @@ export class ModesContentHoverWidget extends ContentHoverWidget {
 			msg.contents
 				.filter(contents => !!contents)
 				.forEach(contents => {
-					const renderedContents = renderMarkedString(contents, {
-						actionCallback: (content) => {
-							this._openerService.open(URI.parse(content)).then(void 0, onUnexpectedError);
-						},
-						codeBlockRenderer: (languageAlias, value): string | TPromise<string> => {
-							// In markdown,
-							// it is possible that we stumble upon language aliases (e.g.js instead of javascript)
-							// it is possible no alias is given in which case we fall back to the current editor lang
-							const modeId = languageAlias
-								? this._modeService.getModeIdForLanguageName(languageAlias)
-								: this._editor.getModel().getModeId();
-
-							return this._modeService.getOrCreateMode(modeId).then(_ => {
-								return `<div class="code">${tokenizeToString(value, modeId)}</div>`;
-							});
-						}
-					});
-
+					const renderedContents = renderMarkedString(this._modeService, this._openerService, contents);
```

Comment:
No, the spec isn't very clear. The first case means that the string represent markdown - all of it, including code blocks but also italic, bullet lists etc. The second case is only about code blocks. So, the first case is a superset of the second case. That is the reason for [this](https://github.com/Microsoft/vscode/blob/master/src/vs/base/browser/htmlContentRenderer.ts#L83) code - it takes a construct from the second case and turns it into the first more stronger case. The existence of the second is sort of legacy, we need to keep it to not break the API but the LSP is more free and should not spell out the second case anymore